src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,306 @@
+                        {
+                            await update.SetAsync(
+                                tx,
+                                "update",
+                                error.Timestamp
+                                );
+                            await tx.CommitAsync();
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogError(ex.Message, "Failed to retrieve previous transaction time.");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex.Message, "Could not create a github issue.");
+            }
+        }
+
+        private async Task<GitHubClient> AuthenticateGitHubClient(string issueRepo)
+        {
+            string gitHubToken = null;
+            IGitHubTokenProvider gitHubTokenProvider = Context.GetService<IGitHubTokenProvider>();
+            long installationId = await Context.GetInstallationId(issueRepo);
+            gitHubToken = await gitHubTokenProvider.GetTokenForInstallationAsync(installationId);
+            Logger.LogInformation("GitHub token acquired for " + issueRepo);
+            Octokit.ProductHeaderValue _product;
+            string version = Assembly.GetExecutingAssembly()
+                .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                .InformationalVersion;
+            _product = new Octokit.ProductHeaderValue("Maestro", version);
+            var client = new GitHubClient(_product);
+            var token = new Credentials(gitHubToken);
+            client.Credentials = token;
+            return client;
+        }
+
+        private async Task CreateOrUpdateGithubIssueAsync(RepositoryBranchUpdateHistoryEntry repositoryBranchUpdateHistory , string issueRepo)
+        {
+            Logger.LogInformation("Something failed in the repository : " + repositoryBranchUpdateHistory.Repository);
+            string fyiHandles = _options.FyiHandle;
+            IReliableDictionary<(string, string), int> gitHubIssueEvaluator =
+            await _stateManager.GetOrAddAsync<IReliableDictionary<(string,string), int>>("gitHubIssueEvaluator");
+            var client = await AuthenticateGitHubClient(issueRepo);
+            StringBuilder description = new StringBuilder("Something failed during dependency update" +
+                Environment.NewLine + Environment.NewLine);
+            switch (repositoryBranchUpdateHistory.Method)
+            {
+                // Only UpdateAssetAsync method has subscriptionId as one of the parameters in the arguments column. Other methods do not have this info.
+                case "UpdateAssetsAsync":
+                    // append to the description only if the subscription is valid else do not create the issue
+                    if (!String.IsNullOrEmpty(UpdateAsyncArgumentsParse(repositoryBranchUpdateHistory.Arguments)))
```

Comment:
We probably shouldn't call the "UpdateAsyncArgumentsParse" method twice.  We should fetch the value into a local, then check that local for null before adding. The method does a fair bit of work, so calling it twice is wasteful.